samples/infiniteStreaming.js

```diff
@@ -0,0 +1,284 @@
+ * Note: Correct microphone settings required: check enclosed link, and make
+ * sure the following conditions are met:
+ * 1. SoX must be installed and available in your $PATH- it can be found here:
+ * http://sox.sourceforge.net/
+ * 2. Microphone must be working
+ * 3. Encoding, sampleRateHertz, and # of channels must match header of audioInput file you're
+ * recording to.
+ * 4. Get Node-Record-lpcm16 https://www.npmjs.com/package/node-record-lpcm16
+ * More Info: https://cloud.google.com/speech-to-text/docs/streaming-recognize
+ * 5. Set streamingLimit in ms. 10000 ms = 10 seconds. Maximum limit should be 1/2 of SpeechAPI Streaming Limit.
+ */
+
+//const encoding = 'LINEAR16';
+//const sampleRateHertz = 16000;
+//const languageCode = 'en-US';
+//const streamingLimit = 10000; //set to low number for demonstration purposes
+
+function infiniteStream(
+  encoding,
+  sampleRateHertz,
+  languageCode,
+  streamingLimit
+) {
+  // [START speech_transcribe_infinite_streaming]
+
+  const chalk = require('chalk');
+  const {Transform} = require('stream');
+
+  // Node-Record-lpcm16
+  const record = require('node-record-lpcm16');
+
+  // Imports the Google Cloud client library (currently, only v1p1beta1 contains result-end-time)
+  const speech = require('@google-cloud/speech').v1p1beta1;
+
+  const client = new speech.SpeechClient();
+
+  const config = {
+    encoding: encoding,
+    sampleRateHertz: sampleRateHertz,
+    languageCode: languageCode,
+  };
+
+  const request = {
+    config,
+    interimResults: true,
+  };
+
+  let recognizeStream = null,
+    restartCounter = 0,
+    audioInput = [],
+    lastAudioInput = [],
+    resultEndTime = 0,
+    isFinalEndTime = 0,
+    finalRequestEndTime = 0,
+    newStream = true,
+    bridgingOffset = 0,
+    lastTranscriptWasFinal = false;
+
+  function startStream() {
+    // Clear current audioInput
+    audioInput = [];
+    // Initiate (Reinitiate) a recognize stream
+    recognizeStream = client
+      .streamingRecognize(request)
+      .on('error', err => {
+        if (err.code === 11) {
+          // uncomment the following code to use timeout error to restart
+          //restartStream();
+        } else {
+          console.error('API request error ' + err);
+        }
+      })
+      .on('data', speechCallback);
+
+    //restart stream when streamingLimit expires
+    setTimeout(restartStream, streamingLimit);
+  }
+
+  const speechCallback = stream => {
+    //convert API result end time from seconds + nanoseconds into milliseconds
+    resultEndTime =
+      stream.results[0].resultEndTime.seconds * 1000 +
+      Math.round(stream.results[0].resultEndTime.nanos / 1000000);
+
+    //calculate correct time based on offset from audio sent twice
+    const correctedTime =
+      resultEndTime - bridgingOffset + streamingLimit * restartCounter;
+
+    process.stdout.clearLine();
+    process.stdout.cursorTo(0);
+    let stdoutText = `\n\nReached transcription time limit, press Ctrl+C\n`;
```

Comment:
Should we remove this sentence? 
Since it is infinite streaming now, the time limit shouldn't matter.